roles/openshift_common/tasks/main.yml

```diff
@@ -34,6 +34,26 @@
       use_manageiq: "{{ openshift_use_manageiq | default(None) }}"
       data_dir: "{{ openshift_data_dir | default(None) }}"
 
+- name: Gather current version of docker
+  command: >
+    rpm -q docker --queryformat '%{version}'
+  register: installed_docker
+  failed_when: false
+  changed_when: false
+  
+- name: Gather latest version of common package
+  shell: >
+    yum list available -e 0 -q "{{ openshift.common.service_type}}" 2>&1 | tail -n +2 | awk '{ print $2 }' | sort -r | tr '\n' ' ' | tail -n 1 | cut -d '.' -f1,2
+  register: latest_package
+  failed_when: false
+  changed_when: false
+
+- name: Downgrade docker to 1.8.2 if we're installing a version older than 3.2
+  command: >
+    {{ ansible_pkg_mgr }} -y downgrade docker-1.8.2
+  notify: restart docker
+  when: " '1.9' in installed_docker.stdout and latest_package.stdout in ['3.0','3.1'] "
+
```

Comment:
I'm not sure we want to handle management of the docker package in openshift_common, instead ensuring the version we expect should probably happen in the docker role, and the openshift_docker_facts role should contain the logic for telling the docker role which version we want installed.